### PR TITLE
040-login-system

### PR DESCRIPTION
**Pull request by MihneaBocaKainos**

### CODE REVIEW

File: src/main/java/com/kainos/ea/controller/UserController.java [hunk 1]
```diff
@@ -33,4 +30,16 @@ public Response createUser(User user) {
          return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
       }
    }
+
+   @POST
+   @Path("/login")
+   @Consumes(MediaType.APPLICATION_JSON)
+   @Produces(MediaType.APPLICATION_JSON)
+   public Response logUser(User user) {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/controller/UserController.java:38]: Move this to its own file i.e LoginController

File: src/main/java/com/kainos/ea/dao/UserDao.java [hunk 2]
```diff
@@ -1,12 +1,20 @@
 package com.kainos.ea.dao;
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/UserDao.java:1]: Rename to RegisterDao

File: src/main/java/com/kainos/ea/model/Encryption.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+
+public class Encryption {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/Encryption.java:16]: Utility class can be `final`

File: src/main/java/com/kainos/ea/model/Encryption.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    public static String decrypt(String strToDecrypt, String secret) {
+        try {
+            Key key = generateKey(secret);
+            Cipher cipher = Cipher.getInstance(ALGO);
+            cipher.init(Cipher.DECRYPT_MODE, key);
+            return new String(cipher.doFinal(Base64.getDecoder().decode(strToDecrypt)));
+        } catch (Exception | EncryptionException e) {
+            System.out.println("Error while decrypting: " + e.toString());
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/Encryption.java:36]: Better to add a logger as this statement only prints to console

File: src/main/java/com/kainos/ea/model/Encryption.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        }
+        return null;
+    }
+
+    private static Key generateKey(String secret) throws EncryptionException {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/Encryption.java:41]: EncryptionException is never thrown so you can remove it

File: src/main/java/com/kainos/ea/model/Encryption.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+
+public class Encryption {
+
+    private static final String ALGO = "AES";
+
+    public static String encrypt(String Data, String secret) throws EncryptionException, NoSuchPaddingException, NoSuchAlgorithmException, IllegalBlockSizeException, BadPaddingException, InvalidKeyException {
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/model/Encryption.java:20]: parameter name should be lower case i.e `data`

**Review by hellofadude — APPROVED** [review]

**Review by asadzafark — APPROVED** [review]

**Pull request merged**